Fix deprecated getEntity (fixes #67)

mikaelwaltersson opened the pull request:
WARNING: DraftEntity.get will be deprecated soon! Please use "contentState.getEntity" instead.

sstur commented on src/stateToHTML.js:
```diff
@@ -157,19 +156,19 @@ class MarkupGenerator {
+    );
+    this.inlineStyles = inlineStyles;
+    this.styleOrder = styleOrder;
+  }
```
What's the difference between the code on the left and the code on the right? If there are no changes can you rebase this PR so they are the same?

mikaelwaltersson replied:
No change really, must be some spacing or something changed just be opening and saving the file in my editor. I couldn't manage to unstage the hunk either using SourceTree.

sstur commented on src/stateToHTML.js:
```diff
@@ -418,6 +417,6 @@ function encodeAttr(text: string): string {
-export default function stateToHTML(content: ContentState, options: ?Options): string {
-  return new MarkupGenerator(content, options).generate();
+export default function stateToHTML(content: ContentState, options: ?Options): string {
+  return new MarkupGenerator(content, options).generate();
```
Also, what is the difference here?

mikaelwaltersson replied:
No change really, must be some spacing or something changed just be opening and saving the file in my editor. I couldn't manage to unstage the hunk either using SourceTree.

sstur commented on src/stateToHTML.js:
```diff
@@ -341,7 +340,7 @@ class MarkupGenerator {
         return content;
       }).join('');
-      let entity = entityKey ? Entity.get(entityKey) : null;
+      let entity = entityKey ? this.contentState.getEntity(entityKey) : null;
```
This is the main change of this PR, right? This fixes the issue?

mikaelwaltersson replied:
Correct, Entity.get is deprecated as of 0.10 and replaced by contentState.getEntity

mikaelwaltersson replied:
https://facebook.github.io/draft-js/docs/v0-10-api-migration.html#content

mikaelwaltersson replied:
I have removed the accidental trailing whitespace changes now.

sstur reviewed (commented):
Thanks for this contribution! I've got a few questions here..

mikaelwaltersson committed:
Fix issue #67
WARNING: DraftEntity.get will be deprecated soon! Please use "contentState.getEntity" instead

gbbr commented:
👍

svnm commented:
Have tested and this works well for me in `Draft 0.10.0` 👍

blade254353074 commented:
How is it now?

sstur reviewed (dismissed):
Thanks for this contribution. My only concern is that this will NOT work with older versions draft-js which do not support `contentState.getEntity()`

We should create a helper function with feature detection:
```
function getEntity(contentState, entityKey) {
  return contentState.getEntity ? contentState.getEntity(entityKey) : Entity.get(entityKey);
}
```
Then you can change line 343 to:
```
let entity = entityKey ? getEntity(this.contentState, entityKey) : null;
```

mikaelwaltersson committed:
Fix Issue #67 WARNING... made fix backward compatible with older versions of draft-js as per sstur's suggestion

mikaelwaltersson commented:
Added your improvements for backward compatibility

sstur commented on src/stateToHTML.js:
```diff
@@ -377,6 +376,10 @@ class MarkupGenerator {
 }
 
+function getEntity(contentState, entityKey) {
+  return contentState.getEntity ? contentState.getEntity(entityKey) : Entity.get(entityKey);
```
`Entity` is not defined, is it?
I think because you removed line 7.

sstur commented:
Please run `npm test` before submitting. This is partly my fault because I should setup TravisCI to automatically run test on all PRs. This is the error:

<img width="1091" alt="screenshot 2017-03-23 12 41 56" src="https://cloud.githubusercontent.com/assets/369384/24233657/3eac4994-0fc6-11e7-8873-fad8146dc721.png">

mikaelwaltersson committed:
Fix issue #67: reversed removal of Entity import

mikaelwaltersson commented:
Sorry for that, weren't able to install all the tools necessary to run the tests and unfortunately I was to busy to look into it any further. 

<img width="352" alt="screen shot 2017-03-23 at 4 59 46 pm" src="https://cloud.githubusercontent.com/assets/1998925/24234021/2eb0c9d8-0fea-11e7-94fb-e31af07a5676.png">

sstur commented:
Thanks!
